src/poetry/plugins/plugin_manager.py

```diff
@@ -80,4 +100,209 @@ def _load_plugin_entry_point(self, ep: metadata.EntryPoint) -> None:
                 "The Poetry plugin must be an instance of Plugin or ApplicationPlugin"
             )
 
-        self.add_plugin(plugin())
+        self._add_plugin(plugin())
+
+
+class ProjectPluginCache:
+    PATH = Path(".poetry") / "plugins"
```

Comment:
Please add writing `.gitginore` file to the `.poetry` directory by default so that we can avoid people committing it to their repositories.

Reply:
done